linode/objbucket/schema_resource.go

```diff
@@ -4,14 +4,18 @@ import "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
 var resourceSchema = map[string]*schema.Schema{
 	"secret_key": {
-		Type:        schema.TypeString,
-		Description: "The S3 secret key to use for this resource. (Required for lifecycle_rule and versioning)",
-		Optional:    true,
+		Type: schema.TypeString,
+		Description: "The S3 secret key to use for this resource. (Required for lifecycle_rule and versioning). " +
+			"If not specified with the resource, the value will be read from provider-level obj_secret_key.",
+		Optional: true,
+		Computed: true,
 	},
 	"access_key": {
-		Type:        schema.TypeString,
-		Description: "The S3 access key to use for this resource. (Required for lifecycle_rule and versioning)",
-		Optional:    true,
+		Type: schema.TypeString,
+		Description: "The S3 access key to use for this resource. (Required for lifecycle_rule and versioning). " +
+			"If not specified with the resource, the value will be read from provider-level obj_access_key.",
+		Optional: true,
+		Computed: true,
```

Comment:
I wonder if it'd make sense to hash these in state like we do for root_pass: https://github.com/linode/terraform-provider-linode/blob/dev/linode/instance/schema_resource.go#L340

Reply:
I changed a bit around with the hashing and find it will add a lot complexity to the obj keys management. The use case for root_pass is a bit simpler, because it's only used while creating and its value in the state is never reused/read. Also it could be more difficult when we're going to migrate these to framework, since the config value doesn't match the state. I've read some issue with StateFunc migration in the Hashicorp discussion.

Instead of hashing these value, how about marking the secret_key as `Sensitive`? It's also what we do in the `objkey` schema: https://github.com/linode/terraform-provider-linode/blob/dev/linode/objkey/framework_resource_schema.go#L32

Reply:
Marking them as sensitive sounds perfect 👍